lib/axios.js

```diff
@@ -27,6 +28,35 @@ function createInstance(defaultConfig) {
     return createInstance(mergeConfig(defaultConfig, instanceConfig));
   };
 
+  // Listen to unhandled rejected axios promises
+  function useDefaultReject(axiosError, unhandledPromiseEvent, promise) {
+    if (
+      isAxiosError(axiosError) &&
+      typeof instance.defaults.defaultReject === 'function' &&
+      // Only execute the defaultReject of the axios instance that was used
+      axiosError.config._instanceId === instance._instanceId
+    ) {
+      promise.catch(function catchUnhandledPromise() {
+        instance.defaults.defaultReject(axiosError);
+        if (typeof unhandledPromiseEvent.preventDefault === 'function') {
+          unhandledPromiseEvent.preventDefault();
+        }
+      });
+    }
+  }
+
+  if (typeof process !== 'undefined' && Object.prototype.toString.call(process) === '[object process]') {
+    // For node use process.addListener
+    process.addListener('unhandledRejection', function handleServerSideUnhandledRejection(e, promise) {
+      useDefaultReject(e, e, promise);
+    });
+  } else if (typeof window !== 'undefined') {
+    // For browsers use window.addEventListener
+    window.addEventListener('unhandledrejection', function handleBrowserSideUnhandledRejection(e) {
```

Comment:
not sure why you use here named function

Reply:
Running tests forces me to do so